File: ChatGPT/python/output/output_23b.py
# ChatGPT-generated solution will go here.
from collections import defaultdict
from itertools import combinations

def main():
    with open("input.txt") as f:
        edges = [line.strip().split('-') for line in f if line.strip()]

    graph = defaultdict(set)
    for a, b in edges:
        graph[a].add(b)
        graph[b].add(a)

    nodes = list(graph.keys())
    max_clique = []

    def is_clique(clique):
        return all(b in graph[a] for a, b in combinations(clique, 2))

    def dfs(clique, candidates):
        nonlocal max_clique
        if len(clique) > len(max_clique):
            max_clique = clique[:]
        for i, node in enumerate(candidates):
            new_clique = clique + [node]
            new_candidates = [n for n in candidates[i+1:] if all(n in graph[x] for x in new_clique)]
            dfs(new_clique, new_candidates)

    dfs([], nodes)
    print(','.join(sorted(max_clique)))

if __name__ == "__main__":
    main()

File: ChatGPT/clojure/output/output_23b.clj
;; ChatGPT-generated solution will go here.
(ns output_23b
  (:require [clojure.string :as str]
            [clojure.set :as set]))

(defn build-graph [edges]
  (reduce (fn [g [a b]]
            (-> g
                (update a (fnil conj #{}) b)
                (update b (fnil conj #{}) a)))
          {}
          edges))

(defn bron-kerbosch
  [graph R P X]
  (lazy-seq
   (if (empty? P)
     (when (empty? X) [R])
     (mapcat (fn [v]
               (bron-kerbosch
                graph
                (conj R v)
                (set/intersection P (graph v))
                (set/intersection X (graph v))))
             P))))

(defn part [lines]
  (let [edges (map #(str/split % #"-") lines)
        graph (build-graph edges)
        nodes (set (keys graph))
        cliques (bron-kerbosch graph #{} nodes #{})
        largest (apply max-key count cliques)]
    (->> largest
         sort
         (str/join ","))))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))
